tests/deep/Tools/Radiation/Models/Cone/Bremsstrahlung/Test_BremsstrahlungScreenedEmission.cpp

```diff
@@ -0,0 +1,221 @@
+{8.723700e+19, 3.280300e+19, 4.708600e+19, 9.909300e+19},
+{9.563800e+19, 5.024300e+19, 4.180200e+19, 7.086400e+19},
+{1.010200e+20, 1.009100e+20, 6.915900e+19, 4.325700e+19, 9.530600e+19},
+{5.424000e+19, 1.004360e+20, 1.331800e+19, 6.324300e+19, 8.165000e+19},
+{2.793000e+19, 4.365300e+19, 2.877100e+19, 4.219300e+19, 5.038600e+19, 6.485700e+19},
+{1.487400e+19, 6.527300e+19, 3.748100e+19, 3.415000e+19, 3.431500e+19, 2.541600e+19}
+};
+
+const slibreal_t Test_BremsstrahlungScreenedEmission::TEST_NR_POW[10][2] = {
+{1,	6.73045973479381e-10},
+{1,	2.01662976950185e-09},
+{2,	1.13293399609858e-09},
+{2,	2.50881731824007e-09},
+{4,	7.60986501112254e-09},
+{4,	5.86883166300790e-09},
+{5,	1.57595925024440e-08},
+{5,	8.02396169262980e-09},
+{6,	4.54479515488073e-09},
+{6,	3.45373484297087e-09}
+};
+
+const slibreal_t Test_BremsstrahlungScreenedEmission::TEST_GAMMA_INTSPEC[10][2] = {
+{1,	    0},
+{5,	    2.15962917566497e-09},
+{10,	5.15999930299167e-09},
+{15,	7.54998734239249e-09},
+{25,	1.10284516497038e-08},
+{30,	1.23582564705472e-08},
+{40,	1.45255528103682e-08},
+{50,	1.62554212133256e-08},
+{75,	1.87447632097682e-08},
+{100,	2.02556589173730e-08}
+};
+
+/**
+ * Check that the formula for total emission
+ * of synchrotron radiation has been implemented
+ * correctly. Tests vs values obtained from same formulas implemented in matlab
+ */
+bool Test_BremsstrahlungScreenedEmission::CheckTotalEmission(const slibreal_t tol) {
+    unsigned int i;
+    Detector *det = GetDetector(0);
+    slibreal_t pwr, corr, Delta;
+
+    for (i = 0; i < NTESTVALUES; i++) {
+	    slibreal_t Input_Z[6] = {TEST_Z[i][0], TEST_Z[i][1], TEST_Z[i][2], TEST_Z[i][3], TEST_Z[i][4], TEST_Z[i][5]};
+        slibreal_t Input_DENS[6] = {TEST_DENS[i][0], TEST_DENS[i][1], TEST_DENS[i][2], TEST_DENS[i][3], TEST_DENS[i][4], TEST_DENS[i][5]};
+        slibreal_t Input_Z0[6] = {TEST_Z0[i][0], TEST_Z0[i][1], TEST_Z0[i][2], TEST_Z0[i][3], TEST_Z0[i][4], TEST_Z0[i][5]};
+        ConeBremsstrahlungScreenedEmission cbse(det, nullptr, TEST_NR_POW[i][INR], Input_Z, Input_Z0, Input_DENS);
+
+        cbse.CalculateTotalEmission();
+        pwr = cbse.GetTotalEmission();
+        corr = TEST_NR_POW[i][IPOWER];
+        
+        Delta = fabs((pwr-corr)/corr);
+
+        if (Delta >= tol) {
+            this->PrintError("Total bremsstrahlung screened emission was not calculated correctly. Delta = %e", Delta);
+            return false;
+        }
+    }
+
+    return true;
+}
+
+/* 
+ * Test so that the spectrum integrates to the correct value
+ */
+bool Test_BremsstrahlungScreenedEmission::CheckSpectrumEmission(const slibreal_t tol) {
+    
+    unsigned int i;
+    Detector *det = GetDetector(50, 1, 50);
+    MagneticFieldAnalytical2D *dummy_mf = GetMagneticField();
+    slibreal_t pwr, corr, Delta;
+
+    const slibreal_t nspecies = TEST_NR_POW[NTESTVALUES-1][INR]; 
+
+    slibreal_t Z[(const unsigned int) nspecies], Z0[(const unsigned int) nspecies], DENS[(const unsigned int) nspecies];
```

Comment:
...and no need for ``(const unsigned int)nspecies`` here then. Just ``nspecies``.